server/src/main/kotlin/suwayomi/tachidesk/manga/impl/Manga.kt

```diff
@@ -284,8 +286,23 @@ object Manga {
         }
     }
 
+    suspend fun getMangaThumbnail(mangaId: Int): Pair<InputStream, String> {
+        val mangaEntry = transaction { MangaTable.select { MangaTable.id eq mangaId }.first() }
+
+        if (mangaEntry[MangaTable.inLibrary]) {
+            return try {
+                ThumbnailDownloadHelper.getImage(mangaId)
+            } catch (_: MissingThumbnailException) {
+                ThumbnailDownloadHelper.download(mangaId)
+                ThumbnailDownloadHelper.getImage(mangaId)
+            }
+        }
+
+        return fetchMangaThumbnail(mangaId)
+    }
+
     private fun clearMangaThumbnailCache(mangaId: Int) {
-        val saveDir = applicationDirs.thumbnailsRoot
+        val saveDir = applicationDirs.tempThumbnailCacheRoot
```

Comment:
We call this when the thumbnail url was changed, so we need to download the new cover, is this properly handled for library manga?

Reply:
it was not - updated